productivity-suite/piercing-library/src/message-bus/message-bus.ts

```diff
@@ -60,6 +60,7 @@ export class GenericMessageBus implements MessageBus {
   }
 
   dispatch(eventName: string, value: JSONValue) {
+    value = JSON.parse(JSON.stringify(value));
```

Comment:
Can you please add a comment to explain what's going on here?

Also could we use https://developer.mozilla.org/en-US/docs/Web/API/structuredClone instead? (structuredClone feels less hacky)